File: adminapp/src/data/repositories/chapterRepository.ts
import axios from 'axios';
import { Chapter } from '../../domain/entities/Chapter';
import { API_URL } from '../../config';

interface ApiResponse<T> {
  success: boolean;
  data: T;
  message?: string;
}

interface PaginatedResponse<T> {
  success: boolean;
  data: T[];
  total: number;
  page: number;
  limit: number;
}

class ChapterRepository {
  private readonly baseUrl = `${API_URL}/chapters`;

  async getChaptersByNovelId(novelId: string, page = 1, limit = 50): Promise<Chapter[]> {
    const response = await axios.get<PaginatedResponse<Chapter>>(`${this.baseUrl}/novel/${novelId}`, {
      params: { page, limit },
    });
    return response.data.data;
  }

  async getChapterById(id: string): Promise<Chapter> {
    const response = await axios.get<ApiResponse<Chapter>>(`${this.baseUrl}/${id}`);
    return response.data.data;
  }

  async createChapter(chapterData: Partial<Chapter>): Promise<Chapter> {
    const response = await axios.post<ApiResponse<Chapter>>(this.baseUrl, chapterData);
    return response.data.data;
  }

  async updateChapter(id: string, chapterData: Partial<Chapter>): Promise<Chapter> {
    const response = await axios.put<ApiResponse<Chapter>>(`${this.baseUrl}/${id}`, chapterData);
    return response.data.data;
  }

  async deleteChapter(id: string): Promise<void> {
    await axios.delete(`${this.baseUrl}/${id}`);
  }
}

export const chapterRepository = new ChapterRepository(); 

File: frontend/lib/domain/repositories/chapter_repository.dart
import '../entities/chapter.dart';

abstract class ChapterRepository {
  Future<List<Chapter>> getChapters(String novelId);
  Future<Chapter> getChapterById(String novelId, String chapterId);
  Future<Chapter> createChapter(String novelId, Chapter chapter);
  Future<Chapter> updateChapter(String novelId, String chapterId, Chapter chapter);
  Future<void> deleteChapter(String novelId, String chapterId);
  Future<void> updateReadingProgress(String novelId, String chapterId);
  Future<String?> getLastReadChapter(String novelId);
} 